migrations/20221020134529_add_saml.up.sql

```diff
@@ -0,0 +1,89 @@
+-- Multi-instance mode (see auth.instances) table intentionally not supported and ignored.
+
+create table if not exists auth.sso_providers (
+	id uuid not null,
+	resource_id text null,
+	created_at timestamptz null,
+	updated_at timestamptz null,
+	primary key (id),
+	constraint "resource_id not empty" check (resource_id = null or char_length(resource_id) > 0)
```

Comment:
if `resource_id` is optional, why can't it be an empty string or null?

Reply:
This means that it has to either be `null` or not empty. It should not be `''`.